From 7ff3fd1403268702df06fe97f353fc8676c15e10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:48:44 +0800
Subject: [PATCH] Revert "USB: composite: Add flag to usb_function to hide its
 interface during enumeration"

This reverts commit ee26eaf1392e4f38ed8d23047a6aa230d5a68188.
---
 drivers/usb/gadget/composite.c | 2 +-
 include/linux/usb/composite.h  | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 0dedc681fc30..7e2dacc7974d 100755
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -267,7 +267,7 @@ static int config_buf(struct usb_configuration *config,
 			descriptors = f->hs_descriptors;
 		else
 			descriptors = f->descriptors;
-		if (f->hidden || !descriptors || descriptors[0] == NULL) {
+		if (!descriptors || descriptors[0] == NULL) {
 			for (; f != config->interface[interfaceCount];) {
 				interfaceCount++;
 				c->bNumInterfaces--;
diff --git a/include/linux/usb/composite.h b/include/linux/usb/composite.h
index 8f33ae3e00f3..4f6bb3d2160e 100644
--- a/include/linux/usb/composite.h
+++ b/include/linux/usb/composite.h
@@ -100,7 +100,6 @@ struct usb_function {
 	struct usb_descriptor_header	**hs_descriptors;
 
 	struct usb_configuration	*config;
-	int				hidden;
 
 	/* REVISIT:  bind() functions can be marked __init, which
 	 * makes trouble for section mismatch analysis.  See if
-- 
2.35.3

